lib/ui/text.dart

```diff
@@ -838,23 +838,42 @@ class TextBox {
 /// For example, when a text position exists at a line break, a single offset has
 /// two visual positions, one prior to the line break (at the end of the first
 /// line) and one after the line break (at the start of the second line). A text
-/// affinity disambiguates between those cases. (Something similar happens with
-/// between runs of bidirectional text.)
+/// affinity disambiguates between those cases.
+///
+/// This affects only line breaks caused by wrapping, not explicit newline
+/// characters. For newline characters, the position is fully specified by the
+/// offset alone, and there is no ambiguity.
+///
+/// TextAffinity also affects bidirectional text at the interface between
+/// LTR and RTL text. Consider the following string, where the lowercase letters
+/// will be displayed as LTR and the uppercase letters RTL: "helloHELLO".  The
+/// string would appear as "helloOLLEH". A text position of 6 would be
+/// ambiguous, since it could be represented as being to the right of the "o"
+/// or to the right of the "H".
```

Comment:
It would be helpful if this paragraph concluded by saying that if TextAffinity was upstream text position 6 ( think you mean 5) would mean for a text cursor.